submissions/TaniaBondarenko/frogger_game/app.js

```diff
@@ -2,62 +2,76 @@ const borders = {
     rightBorder: 400,
     leftBorder: 5,
     upBorder: 410,
-    downBorder: 70
+    downBorder: 5
 };
+const someGap = 10;
 
-const Enemy = function (x, y) {
-    this.sprite = 'images/enemy-bug.png';
-    this.x = x;
-    this.y = y;
-    this.speed = 75;
-
-    this.render = function () {
+class GameSprite {
+    constructor (x, y,sprite)  {
+        this.sprite = sprite;
+        this.x = x;
+        this.y = y;
+    }
+    render = function () {
         ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
     };
+}
+
+class Enemy extends GameSprite {
+    constructor(x, y, sprite = "images/enemy-bug.png", speed = "75") {
+        super(x, y, sprite);
+        this.x = x;
+        this.y = y;
+        this.speed = speed;
+        this.sprite = sprite;
+    }
+    render(){};
 };
 
     Enemy.prototype.update = function (dt) {
         this.x += this.speed * dt;
+        const addSpeed = 40;
         if (this.x >= 500) {
-            this.x = -10;
-            this.speed = Math.floor(40 + Math.random() * 100);
+        this.x = -someGap;
+        this.speed = Math.floor(addSpeed + Math.random() * 100);
         }
     };
 
 this.checkCollisions = function () {
     allEnemies.forEach(function (enemy) {
+        const collision_distance = 50;
         let diffX = Math.abs(enemy.x - player.x);
         let diffY = Math.abs(enemy.y - player.y);
         let distance = Math.sqrt(Math.pow(diffX, 2) + Math.pow(diffY, 2));
-        if (distance <= 50) {
+        if (distance <= collision_distance) {
             alert(`Oops... \uD83D\uDE3F Try Again`);
             player.initialCoordinates();
         } else if (player.y === borders.downBorder) {
-            player.y = 60;
+            player.y = someGap;
             setTimeout(function () { alert(`\uD83D\uDE38 Congratulations! You won! \uD83D\uDE38 `); }, 100);
-            setTimeout(function () { player.initialCoordinates(); }, 50);
+            setTimeout(function () { player.initialCoordinates(); }, 200);
         };
     });
 };
 
-const Player = function (x, y) {
-    this.sprite = "images/char-cat-girl.png";
-    this.x = x;
-    this.y = y;
-    this.stepX = 98;
-    this.stepY = 85;
-    this.initialCoordinates = function () {
-        this.x = 202; this.y = 400;
+class Player extends GameSprite {
+    constructor(x = 202, y = 400, sprite = "images/char-cat-girl.png") {
+        super(x, y, sprite);
+        this.sprite = sprite;
+        this.stepX = 98;
+        this.stepY = 85;
     }
 
-    this.update = function () {
-    };
+    render() { };
```

Comment:
we don't need this redefinition too... we already have render() method that defined in parent class 'GameSprite'
better to remove this line too...

Reply:
done